components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/dao/impl/ApplicationDAOImpl.java

```diff
@@ -3747,6 +3773,54 @@ public int getCountOfApplications(String filter) throws IdentityApplicationManag
         return count;
     }
 
+    /**
+     * Retrieve application basic information using the sp metadata property key and value.
+     *
+     * @param key Name of the sp metadata property key
+     * @param value Value of the sp metadata property value
+     * @return
+     * @throws IdentityApplicationManagementException
+     */
+    @Override
+    public ApplicationBasicInfo[] getApplicationBasicInfoBySPProperty(String key, String value)
+            throws IdentityApplicationManagementException {
+
+        int tenantID = CarbonContext.getThreadLocalCarbonContext().getTenantId();
+        if (log.isDebugEnabled()) {
+            log.debug("Getting the all applications for the tenant: " + tenantID);
+        }
+        Connection connection = IdentityDatabaseUtil.getDBConnection(false);
+        PreparedStatement getAppNamesStmt = null;
+        ResultSet appNameResultSet = null;
+        ArrayList<ApplicationBasicInfo> appInfo = new ArrayList<ApplicationBasicInfo>();
+
+        try {
```

Comment:
Shall we use try with resource, making connection and preparedStatement resources which auto closable at the end of usage?